frontend/src/App.tsx

```diff
@@ -989,16 +1020,26 @@ export class App extends PureComponent<Props, State> {
       return
     }
 
+    const { currentPageScriptHash } = this.state
     const { basePath } = baseUriParts
     const queryString = this.getQueryString()
+    let pageName = ""
+
+    if (pageScriptHash) {
```

Comment:
why check for this? Can't we just start from 
`if (currentPageScriptHash)`

Reply:
If the user is switching pages, you need to use the hash of the page they're asking to run and not the one they're currently on